### DISCUSSION TITLE
Welcome to respeaker-websockets Discussions!

### DISCUSSION THREAD

**Discussion by sskorol** [sskorol/respeaker-websockets#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by afiq234** [comment]
i want to ask a question what's the difference between respeaker and manually created voice assistant using mic and speaker for raspberry pi. I also wondered when there's a lot of tutorials that teach how to create voice assistant without using respeaker. I want to know what make respeaker from seeed studio so special

**Reply by sskorol** [reply]
@afiq234 with the common mic you can interact with your assistant at a very short distance only. It's not a far-field device in 99% of cases. People often use either a headset or some cheap desktop mics (or even integrated into a laptop or PC). But if you try to move couple of meters away from your mic, it won't hear you anymore.

Now imaging one of the well-known smart speakers, e.g. Alexa or Google Home. They can hear and understand you even if you talk from the opposite side of the room. Moreover, they can hear you even if there's bad background noise or when music is playing from the speaker itself. They can wake by a special wake-word, focus on a person who called them and apply some complicated DSP algorithms against the input audio stream: noise suppression, automatic gain control, acoustic echo cancellation, beamforming, etc. That's very important to supply clean data to ASR and NLU engines. Otherwise, they won't be able to hear and understand you.

Even when you build your custom assistant, it's required to supply good and clean voice data for further transcribing, extracting entities, predicting intents, etc. That's why a good microphone is very important. Also, note that these smart speakers (like Alexa) have several bundled microphones - so-called microphone arrays. With a single microphone, it's impossible to split several audio sources and focus on a particular beam, e.g. when you and your TV are talking simultaneously.

So what does Respeaker offer? First of all, it's a microphone array with 6 bundled mics. Secondly, you may program it to apply some ready-to-use algorithms (see above). They are closed-source, but on the other hand, you won't find anything better in open-source. Generally speaking, it's not perfect and won't give you the same quality as Alexa or Google. But as for me, it's currently the best devkit that you can find for the offline ASR tasks.
